File: src/dataset/cpp/bob_exercism-tqa236/claude/ClaudeSonnet4_bob_v2.cpp
#include "bob.h"

std::string bob::hey(std::string text) {
  bool no_lowercase = true;
  bool has_uppercase = false;
  bool is_question = false;
  bool is_empty = true;
  
  for (const char c : text) {
    if (!std::isspace(c)) {
      is_empty = false;
      if (c == '?') {
        is_question = true;
      } else {
        is_question = false;
      }
    }
    if (std::islower(c)) {
      no_lowercase = false;
    } else if (std::isupper(c)) {
      has_uppercase = true;
    }
  }
  
  if (is_empty) {
    return "Fine. Be that way!";
  }
  
  bool is_yelling = no_lowercase && has_uppercase;
  if (is_yelling) {
    return is_question ? "Calm down, I know what I'm doing!" : "Whoa, chill out!";
  }
  
  return is_question ? "Sure." : "Whatever.";
}

File: src/dataset/javascript/bob_exercism-ThomasZumsteg/claude/ClaudeSonnet4_bob_v2.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
	// Cache trimmed input to avoid multiple operations
	var trimmed = input.trim();
	
	// Nothing (empty or whitespace only)
	if (!trimmed)
		return "Fine. Be that way!";
	
	var isQuestion = trimmed.charAt(trimmed.length - 1) === '?';
	var hasLetters = /[A-Za-z]/.test(trimmed);
	var isUpperCase = hasLetters && trimmed === trimmed.toUpperCase();
	
	// Shouting (all upper case with letters)
	if (isUpperCase)
		return "Whoa, chill out!";
	
	// Question (ends with a question mark)
	if (isQuestion)
		return "Sure.";
	
	// Default
	return "Whatever.";
};

module.exports = Bob;